### PR TITLE
Knative operator job added for ppc64le 

### PR DESCRIPTION
**Pull request by valen-mascarenhas14**
With the Knative community's recent decision to disable jobs for P & Z architectures in the upstream Prow CI ([knative/infra#497](https://github.com/knative/infra/pull/497)), we are migrating the Knative CI jobs to a local Prow server to ensure continued support for Power architecture.

Here's the Slack thread for discussions & planning for this migration ([discussion thread](https://ibm-cloudplatform.slack.com/archives/G01CH7AC8H1/p1723184215935419)) to align with the community's needs while addressing the specific requirements for Power .

This PR marks the first step in a broader effort to ensure that Knative's CI infrastructure remains robust and inclusive for power arch. 

**Key Actions:**
]Initiated the migration process to our local Prow CI setup to maintain CI coverage for Power architecture, starting with the knative/operator tests, which are relatively straightforward to implement.

Further migrations will follow as we continue

### CODE REVIEW

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (mkumatag) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:54]: what are these scripts? right now in this repo, we are not dealing any of the code in github.ibm.com.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (Rajalakshmi-Girish) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:55]: Use `pushd` and `popd` while switching directories.

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (Rajalakshmi-Girish) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:62]: You must have to run the `kubetest2 --down` command from the same directory where you ran the `--up` from.
Also, did you do a `test-pj` run of this job yaml against any other local k8s cluster?

File: config/prow/config.yaml
Comment (Rajalakshmi-Girish) [inline config/prow/config.yaml:274]: Get rid of this ` no newline at end of file`

**Review by Rajalakshmi-Girish — CHANGES_REQUESTED** [review]

**Comment by mkumatag** [comment]
I see no activities here, @valen-mascarenhas14 any plan for completion?

**Comment by Rajalakshmi-Girish** [comment]
> Quote reply

Seems like they have created a RTC task to complete OSSC approval process for migrating https://github.ibm.com/ppc64le-automation/knative-upstream-ci to public as per our suggestion.

**Comment by valen-mascarenhas14** [comment]
Hi @mkumatag , @Rajalakshmi-Girish ,
We have created an RTC task for OSSC approval to externalize the (https://github.ibm.com/ppc64le-automation/knative-upstream-ci) repo. While we await the approval, we are conducting local tests on the other components. Once the approval is granted, we'll proceed with raising the necessary PRs for the other products.

**Comment by Rajalakshmi-Girish** [comment]
@mkumatag I have created copied their enterprise repo to https://github.com/ppc64le-cloud/knative-upstream-ci as they mentioned to have acquired OSSC approvals to do so.

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (valen-mascarenhas14) [reply config/jobs/periodic/knative/operator/main/operator-main.gen.yaml]: Removed the above config

File: config/prow/config.yaml
Comment (valen-mascarenhas14) [reply config/prow/config.yaml]: done

File: config/prow/config.yaml
Comment (mkumatag) [inline config/prow/config.yaml:248]: Try avoid using the internal registry here, we usually dont maintain any credentials which talks to IBM artifacts.

File: config/prow/config.yaml
Comment (mkumatag) [inline config/prow/config.yaml:238]: why do we need GITHUB_TOKEN?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: config/prow/config.yaml
Comment (valen-mascarenhas14) [reply config/prow/config.yaml]: Removed the GITHUB_TOKEN.

File: config/prow/config.yaml
Comment (ghatwala) [reply config/prow/config.yaml]: @mkumatag  - can utilize icr.io registry in above use case , would it be fine to use that ?

Comment (mkumatag) [reply config/prow/config.yaml]: If it is only for the test then use the GitHub registry itself or quay which doesn't include any cost

Comment (valen-mascarenhas14) [reply config/prow/config.yaml]: @mkumatag  we use the jfrog/icr registry for storing the ppc64le specific images which aren't multi arch & the images built during test runtime.

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (mkumatag) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:58]: @Rajalakshmi-Girish TODO: make sure you add a pvsadm cleanup job for this workspace

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (mkumatag) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:51]: this can be moved to `extra_refs` section above

File: config/prow/config.yaml
Comment (mkumatag) [inline config/prow/config.yaml:245]: I still feel that you folks are just wasting resources by pushing these images to ICR if not been used by any endusers, see if you can use any other repos like quay/ghcr

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Comment by mkumatag** [comment]
- fix the yamlint errors
- setup a meeting if needed to discuss on the image registry plans.

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (Rajalakshmi-Girish) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:9]: Have forgotten to change the org to `ppc64le-cloud` ?

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (Rajalakshmi-Girish) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:15]: The default value of `workdir` is false. Needn't mention.

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (Rajalakshmi-Girish) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:23]: @valen-mascarenhas14 Any reference/analysis done on why 4000m Guaranteed CPUs needed?

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (Rajalakshmi-Girish) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:56]: Hope the Junit test result part is taken care of. (@valen-mascarenhas14 we discussed in initial days on teams)

File: config/prow/config.yaml
Comment (Rajalakshmi-Girish) [inline config/prow/config.yaml:262]: Cosmetic suggestion, you may remove the suffix `-volume` to names.

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (Rajalakshmi-Girish) [inline config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:65]: Hope this env is required internally and is not redundant.

**Review by Rajalakshmi-Girish — CHANGES_REQUESTED** [review]

File: config/prow/config.yaml
Comment (valen-mascarenhas14) [reply config/prow/config.yaml]: Removed the `--volume` suffix from the secrets

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (valen-mascarenhas14) [reply config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:56]: Yes, it's taken care of

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (valen-mascarenhas14) [reply config/jobs/periodic/knative/operator/main/operator-main.gen.yaml]: Modified it to 1500m

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (valen-mascarenhas14) [reply config/jobs/periodic/knative/operator/main/operator-main.gen.yaml]: changed it

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (valen-mascarenhas14) [reply config/jobs/periodic/knative/operator/main/operator-main.gen.yaml]: changed the org

File: config/jobs/periodic/knative/operator/main/operator-main.gen.yaml
Comment (valen-mascarenhas14) [reply config/jobs/periodic/knative/operator/main/operator-main.gen.yaml:65]: Yes, this env variable is required .

**Comment by Rajalakshmi-Girish** [comment]
I shall deploy the required secrets on the IKS cluster. Will submit PR to secrets repo too.
@mkumatag ^^

**Review by mkumatag — APPROVED** [review]
/lgtm
/hold

@Rajalakshmi-Girish remove hold whenever you are comfortable.

**Comment by Rajalakshmi-Girish** [comment]
@valen-mascarenhas14 I did a test run to confirm if everything was fine.
The tests results doesn't seem to appear in the JUnit tab as it should be.

Please see https://prow.ppc64le-cloud.cis.ibm.net/view/s3/ppc64le-prow-logs/logs/test-knative-operator-main-periodic/1853481276925808640.

The Logs show that junit xml file is copied to `//home/prow/go/src/github.com/ppc64le-cloud/knative-upstream-ci/knative-1730739627/artifacts`..In our jobs, I see ARTIFACTS path to be `/logs/artifacts`. In your logic to copy the xml files, did you use $ARTIFACTS?

image

**Comment by valen-mascarenhas14** [comment]
> @valen-mascarenhas14 I did a test run to confirm if everything was fine. The tests results doesn't seem to appear in the JUnit tab as it should be.
> 
> Please see https://prow.ppc64le-cloud.cis.ibm.net/view/s3/ppc64le-prow-logs/logs/test-knative-operator-main-periodic/1853481276925808640.
> 
> The Logs show that junit xml file is copied to `//home/prow/go/src/github.com/ppc64le-cloud/knative-upstream-ci/knative-1730739627/artifacts`..In our jobs, I see ARTIFACTS path to be `/logs/artifacts`. In your logic to copy the xml files, did you use $ARTIFACTS?
> 
> image

@Rajalakshmi-Girish  I've removed the  redundant ARTIFACTS directory export from the job

**Comment by Rajalakshmi-Girish** [comment]
> @Rajalakshmi-Girish I've removed the redundant ARTIFACTS directory export from the job

Yes the Junit tab is showing up tests properly after your above change:
image

**Comment by Rajalakshmi-Girish** [comment]
@valen-mascarenhas14 one last thing, Can you please squash all the 11 commits in this PR to one commit?
That is the usual practice we follow.

**Review by Rajalakshmi-Girish — APPROVED** [review]
/lgtm

**Pull request merged**
